apps/server/src/modules/idp-console/idp-sync-console.ts

```diff
@@ -0,0 +1,50 @@
+import { Console, Command } from 'nestjs-console';
+import { ConsoleWriterService } from '@infra/console';
+import { SynchronizationUc } from '@modules/synchronization';
+import { UsersSyncOptions, SystemType } from './interface';
+
+@Console({
+	command: 'sync',
+	description: 'Console providing an access to the IDP-provisioned data synchronization operations.',
+})
+export class IdpSyncConsole {
+	constructor(private consoleWriter: ConsoleWriterService, private synchronizationUc: SynchronizationUc) {}
+
+	@Command({
+		command: 'users',
+		description: 'Synchronize IDP-provisioned users.',
+		options: [
+			{
+				flags: '-st, --systemType <value>',
+				description: 'Type of a synchronized system.',
+				required: true,
+			},
+			{
+				flags: '-si, --systemId <value>',
+				description: 'ID of a synchronized system.',
+				required: true,
+			},
+		],
+	})
+	async users(options: UsersSyncOptions): Promise<void> {
+		if (options.systemType !== SystemType.MOIN_SCHULE) {
+			throw new Error(`invalid system type (currently the only supported system type is "${SystemType.MOIN_SCHULE}")`);
+		}
+
+		this.consoleWriter.info(
+			JSON.stringify({
+				message: 'starting synchronization',
+				options,
+			})
+		);
+
+		await this.synchronizationUc.updateSystemUsersLastSyncedAt(options.systemId);
```

Comment:
generally usecases should never be called from outside their modules. Since they are part of an api Module, that would mean importing that API module, wich has a lot of implications for your deployment (like increasing the API of your deployment). Generally, our console modules include their own usecases, likely this one should be part of the idp-console, accessing a SynchronisationService that is published by the SynchronisationModule.

Reply:
Ok I'll do that as it will also fix this issue https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859#discussion_r1556015029 .

Reply:
Done in https://github.com/hpi-schul-cloud/schulcloud-server/pull/4859/commits/c79e0170a2cf8e9cd4f1de410d4373565276b5af .